tests/foreman/cli/test_classparameters.py

```diff
@@ -219,56 +406,26 @@ def test_negative_update_parameter_type(self):
 
         1.  Parameter not updated with string type for invalid value.
         2.  Error raised for invalid default value.
-
-        @caseautomation: notautomated
-        """
-
-    @run_only_on('sat')
-    @stubbed()
-    @tier3
-    def test_positive_validate_puppet_default_value(self):
-        """Validation doesn't works on puppet default value.
-
-        @id: ca69f1e8-d383-4563-a302-9758ffae1129
-
-        @steps:
-
-        1.  Override the parameter.
-        2.  Set puppet default value to 'Use Puppet Default'.
-        3.  Validate this value under section 'Optional Input Validator'.
-
-        @assert: Validation shouldn't work with puppet default value.
-
-        @caseautomation: notautomated
-
-        @CaseLevel: System
         """
+        sc_param_id = self.sc_params_ids_list.pop()
+        for test_data in invalid_sc_parameters():
+            with self.subTest(test_data):
+                with self.assertRaises(CLIReturnCodeError):
+                    SmartClassParameter.update({
+                        'parameter-type': test_data['sc_type'],
+                        'default-value': test_data['value'],
+                        'id': sc_param_id,
+                        'override': 1,
+                    })
+                sc_param = SmartClassParameter.info({
+                    'puppet-class': 'ntp',
+                    'id': sc_param_id,
+                })
+                self.assertNotEqual(
+                    sc_param['default-value'], test_data['value'])
```

Comment:
I think comparison is not happening properly here in case of boolean, array, json and yaml. We will miss an actual failure here. Good to follow the 'test_positive_update_parameter_type' testcase approach.